### ISSUE TITLE
"npm exec -- renovate" by the server uses the wrong node/npm version after node tool installs

### ISSUE THREAD

**Issue by essh** [mend/renovate-ce-ee#378]
When using Renovate CE 6.x (either the `full` or `slim` images) it's possible for the default `node` & `npm` version on the container filesystem to be updated when a different version of `node` is installed by `renovate` during a run (i.e. in relation to repo configuration like `engines` etc). This results in changes to the `/usr/local/bin/npm` file so that future `npm exec -- renovate` calls by the server process either produce warnings or fail depending on the degree of mis-match between the version that has now become the default.

Processes on a clean container before any `node` version change:
```
  PID TTY      STAT   TIME COMMAND
    1 ?        Ss     0:00 dumb-init -- node src/community.js
    7 ?        Ss     0:00 /bin/bash /usr/local/bin/node src/community.js
    8 ?        Sl     0:04 /opt/containerbase/tools/node/18.18.0/bin/node --use-openssl-ca src/community.js
   78 pts/0    Ss     0:00 /bin/bash
  156 ?        Ss     0:00 /bin/sh -c npm exec -- renovate
  157 ?        S      0:00 /bin/bash /usr/local/bin/npm exec -- renovate
  158 ?        S      0:00 /bin/bash /usr/local/bin/node /opt/containerbase/tools/npm/10.1.0/bin/npm exec -- renovate
  159 ?        Sl     0:00 npm exec renovate
  170 ?        S      0:00 sh -c renovate
  171 ?        S      0:00 /bin/bash /usr/local/bin/node /usr/src/app/node_modules/.bin/renovate
  172 ?        Sl     0:08 /opt/containerbase/tools/node/18.18.0/bin/node --use-openssl-ca /usr/src/app/node_modules/.bi
  243 pts/0    R+     0:00 ps ax
```

`/usr/local/bin/npm` before a `node` version change:
```
#!/bin/bash

if [[ -z "${CONTAINERBASE_ENV+x}" ]]; then
  . /usr/local/etc/env
fi
/opt/containerbase/tools/npm/10.1.0/bin/npm "$@
```

Processes when `node`/`npm` have been changed during a `renovate` run (note renovate is executing against a different version of `node` that now produces warnings or errors):
```
  PID TTY      STAT   TIME COMMAND
    1 ?        Ss     0:00 dumb-init -- node src/community.js
    7 ?        Ss     0:00 /bin/bash /usr/local/bin/node src/community.js
    8 ?        Sl     0:06 /opt/containerbase/tools/node/18.18.0/bin/node --use-openssl-ca src/community.js
   78 pts/0    Ss     0:00 /bin/bash
 1788 ?        Ss     0:00 /bin/sh -c npm exec -- renovate
 1789 ?        S      0:00 /bin/bash /usr/local/bin/npm exec -- renovate
 1790 ?        S      0:00 /bin/bash /usr/local/bin/node /opt/containerbase/tools/node/20.9.0/bin/npm exec -- renovate
 1791 ?        Sl     0:00 npm exec renovate
 1802 ?        S      0:00 sh -c renovate
 1803 ?        S      0:00 /bin/bash /usr/local/bin/node /usr/src/app/node_modules/.bin/renovate
 1804 ?        Sl     0:23 /opt/containerbase/tools/node/20.9.0/bin/node --use-openssl-ca /usr/src/app/node_modules/.bin
 1998 ?        Ss     0:00 /bin/sh -c ./gradlew wrapper --gradle-distribution-url https://services.gradle.org/distributi
 1999 ?        S      0:00 /bin/bash /usr/local/bin/java -Xmx64m -Xms64m -Dorg.gradle.parallel=true -Dorg.gradle.configu
 2015 ?        Sl     0:04 /opt/containerbase/tools/java/17.0.9+9/bin/java -Xmx64m -Xms64m -Dorg.gradle.parallel=true -D
 2050 ?        Ssl    1:45 /opt/containerbase/tools/java/17.0.9+9/bin/java --add-opens=java.base/java.util=ALL-UNNAMED -
 2221 pts/0    R+     0:00 ps ax
```

`/usr/local/bin/npm` after a `node` version change:
```
#!/bin/bash

if [[ -z "${CONTAINERBASE_ENV+x}" ]]; then
  . /usr/local/etc/env
fi
/opt/containerbase/tools/node/20.9.0/bin/npm "$@"
```

Some example warnings & errors observed depending on the version of `node`/`npm/ that ends up being used:
```
renovate-78f6644b98-zfgvm renovate {"name":"renovate","hostname":"renovate-78f6644b98-zfgvm","pid":8,"level":20,"logContext":"QUwqrv99NlbRfsS1h-akW","cmd":"npm exec -- renovate","msg":"Executing Renovate run command","time":"2023-10-26T22:52:07.109Z","v":0}
renovate-78f6644b98-zfgvm renovate
renovate-78f6644b98-zfgvm renovate Usage: npm <command>
renovate-78f6644b98-zfgvm renovate
renovate-78f6644b98-zfgvm renovate where <command> is one of:
renovate-78f6644b98-zfgvm renovate     access, adduser, audit, bin, bugs, c, cache, ci, cit,
renovate-78f6644b98-zfgvm renovate     clean-install, clean-install-test, completion, config,
renovate-78f6644b98-zfgvm renovate     create, ddp, dedupe, deprecate, dist-tag, docs, doctor,
renovate-78f6644b98-zfgvm renovate     edit, explore, fund, get, help, help-search, hook, i, init,
renovate-78f6644b98-zfgvm renovate     install, install-ci-test, install-test, it, link, list, ln,
renovate-78f6644b98-zfgvm renovate     login, logout, ls, org, outdated, owner, pack, ping, prefix,
renovate-78f6644b98-zfgvm renovate     profile, prune, publish, rb, rebuild, repo, restart, root,
renovate-78f6644b98-zfgvm renovate     run, run-script, s, se, search, set, shrinkwrap, star,
renovate-78f6644b98-zfgvm renovate     stars, start, stop, t, team, test, token, tst, un,
renovate-78f6644b98-zfgvm renovate     uninstall, unpublish, unstar, up, update, v, version, view,
renovate-78f6644b98-zfgvm renovate     whoami
renovate-78f6644b98-zfgvm renovate
renovate-78f6644b98-zfgvm renovate npm <command> -h  quick help on <command>
renovate-78f6644b98-zfgvm renovate npm -l            display full usage info
renovate-78f6644b98-zfgvm renovate npm help <term>   search for help on <term>
renovate-78f6644b98-zfgvm renovate npm help npm      involved overview
renovate-78f6644b98-zfgvm renovate
renovate-78f6644b98-zfgvm renovate Specify configs in the ini-formatted file:
renovate-78f6644b98-zfgvm renovate     /home/ubuntu/.npmrc
renovate-78f6644b98-zfgvm renovate or on the command line via: npm <command> --key value
renovate-78f6644b98-zfgvm renovate Config info can be viewed via: npm help config
renovate-78f6644b98-zfgvm renovate
renovate-78f6644b98-zfgvm renovate npm@6.14.13 /opt/containerbase/tools/node/14.17.0/lib/node_modules/npm
renovate-78f6644b98-zfgvm renovat
```

```
renovate-5fbf954974-q8hgs renovate {"name":"renovate","hostname":"renovate-5fbf954974-q8hgs","pid":7338,"level":40,"logContext":"fY60dmp8zJdhuHITrb-Qc","err":{"code":"ERR_DLOPEN_FAILED","message":"The module '/usr/src/app/node_modules/renovate/node_modules/re2/build/Release/re2.node'\nwas compiled against a different Node.js version using\nNODE_MODULE_VERSION 108. This version of Node.js requires\nNODE_MODULE_VERSION 115. Please try re-compiling or re-installing\nthe module (for instance, using `npm rebuild` or `npm install`).","stack":"Error: The module '/usr/src/app/node_modules/renovate/node_modules/re2/build/Release/re2.node'\nwas compiled against a different Node.js version using\nNODE_MODULE_VERSION 108. This version of Node.js requires\nNODE_MODULE_VERSION 115. Please try re-compiling or re-installing\nthe module (for instance, using `npm rebuild` or `npm install`).\n    at Object.Module._extensions..node (node:internal/modules/cjs/loader:1327:18)\n    at Module.load (node:internal/modules/cjs/loader:1091:32)\n    at Function.Module._load (node:internal/modules/cjs/loader:938:12)\n    at Module.require (node:internal/modules/cjs/loader:1115:19)\n    at require (node:internal/modules/helpers:130:18)\n    at Object.<anonymous> (/usr/src/app/node_modules/renovate/node_modules/re2/re2.js:3:13)\n    at Module._compile (node:internal/modules/cjs/loader:1241:14)\n    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1295:10)\n    at Module.load (node:internal/modules/cjs/loader:1091:32)\n    at Function.Module._load (node:internal/modules/cjs/loader:938:12)\n    at Module.require (node:internal/modules/cjs/loader:1115:19)\n    at require (node:internal/modules/helpers:130:18)\n    at re2 (/usr/src/app/node_modules/renovate/lib/expose.cjs:14:10)\n    at Object.<anonymous> (/usr/src/app/node_modules/renovate/lib/util/regex.ts:13:20)\n    at Module._compile (node:internal/modules/cjs/loader:1241:14)\n    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1295:10)\n    at Module.load (node:internal/modules/cjs/loader:1091:32)\n    at Function.Module._load (node:internal/modules/cjs/loader:938:12)\n    at Module.require (node:internal/modules/cjs/loader:1115:19)\n    at require (node:internal/modules/helpers:130:18)"},"msg":"RE2 not usable, falling back to RegExp","time":"2023-10-26T23:56:44.397Z","v":0}
```

**Comment by Gabriel-Ladzaretti** [comment]
Thank you for reporting. A fix will be deployed soon.

**Comment by Gabriel-Ladzaretti** [comment]
addressed in `v6.6.0`.
